tests/Microsoft.Bot.Builder.TestBot/Dialogs/BookingDialog.cs

```diff
@@ -58,7 +58,7 @@ private async Task<DialogTurnResult> FinalActionAsync(WaterfallStepContext stepC
             if ((bool)stepContext.Result)
             {
                 // Pull the booking details from the waterfall state.
-                var bookingDetails = (BookingDetails)stepContext.Values["BookingInfo"];
+                var bookingDetails = stepContext.Values.MapValueTo<BookingDetails>("BookingInfo");
```

Comment:
This shows why making MemoryStorage default to None would break other bots. If we decide to default to All as requested before, we should update the samples to use a MemoryStorage that uses a constructor that sets the value to None and uses this helpers.